superchain/superchain.go

```diff
@@ -499,6 +499,10 @@ func unMarshalSuperchainConfig(data []byte, s *SuperchainConfig) error {
 	return yaml.Unmarshal(data, temp)
 }
 
+// Superchain represents a set of OP Chains which must all upgrade together.
```

Comment:
Hmm this currently is not true. Eventually it will be true for premier chains, and maybe for standard chains, but not for frontier chains.

Reply:
Interesting. What's the right way to think about this struct? 

If we were trying to make it true... Would each Frontier Chain get its own `Superchain`?

Reply:
I'm not sure which is why I did not propose a better phrasing lol 😅 Deferring to @geoknee here since he's more familiar with the code and where that struct is used.

Reply:
I think "superchain" and "supechain target" are used interchangeably. See this PR https://github.com/ethereum-optimism/superchain-registry/pull/186/files.

I think this is an example of where the data structures and terminology was put in place _before_ we got the "new" definitions of standard/frontier/premier, so back when we had a somewhat stricter idea of what a superchain was.

It's still a useful struct for defining sets of chains
In terms of how the struct is actually used there are a few places:
1.  The superchain-wide contract versions check. When we want to loop over each superchain and check the version of the superchain-wide contracts, we can loop over the `Superchains` object and receive a `Superchain` struct inside the loop and make our checks. 

https://github.com/ethereum-optimism/superchain-registry/blob/db623716fc8b7632228b3e8ab0829b05a06a7640/validation/superchain-version_test.go#L50-L51

2. It is exported from the Go module, and used in `op-node` and `op-chain-ops` (for the upgrades). You can search for `superchain.Superchain` to see where. 

For e.g. Frontier chains where we don't expect them to upgrade together, we could consider separating them from the idea of being "in a superchain". This could help reduce confusion, and it might also have a nice side effect of avoiding bloating the downstream modules with a bunch of chains which are mostly irrelevant to them.  See also https://github.com/ethereum-optimism/superchain-registry/issues/151.